From 0c7b073b4697fc71c70997741b2faa440449dd02 Mon Sep 17 00:00:00 2001
From: CraigX Kewley <craigx.kewley@intel.com>
Date: Wed, 8 Mar 2017 15:57:47 +0000
Subject: [PATCH 048/120] ASoC: Intel: Boards: Set TDF8532 constraints

BE DAI constraints don't affect FE constraints. Add a startup callback
to explicitly set constraints for TDF8532 FE.

Signed-off-by: CraigX Kewley <craigx.kewley@intel.com>
---
 sound/soc/intel/boards/bxt_gpmrb.c | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/sound/soc/intel/boards/bxt_gpmrb.c b/sound/soc/intel/boards/bxt_gpmrb.c
index a8386bf..8e816d5 100644
--- a/sound/soc/intel/boards/bxt_gpmrb.c
+++ b/sound/soc/intel/boards/bxt_gpmrb.c
@@ -27,6 +27,7 @@
 
 #define CHANNELS_MONO 1
 #define CHANNELS_STEREO 2
+#define CHANNELS_QUAD 4
 #define CHANNELS_EIGHT 8
 
 static struct snd_soc_dai_link broxton_gpmrb_dais[];
@@ -208,6 +209,35 @@ static int broxton_gpmrb_hdmi_startup(struct snd_pcm_substream *substream)
 	.startup = broxton_gpmrb_hdmi_startup,
 };
 
+static int broxton_gpmrb_tdf8532_startup(struct snd_pcm_substream *substream)
+{
+	int ret;
+
+	ret = snd_pcm_hw_constraint_single(substream->runtime,
+			SNDRV_PCM_HW_PARAM_CHANNELS, CHANNELS_QUAD);
+
+	if (ret < 0)
+		goto out;
+
+	ret = snd_pcm_hw_constraint_single(substream->runtime,
+			SNDRV_PCM_HW_PARAM_RATE, 48000);
+
+	if (ret < 0)
+		goto out;
+
+	ret = snd_pcm_hw_constraint_mask64(substream->runtime,
+			SNDRV_PCM_HW_PARAM_FORMAT, SNDRV_PCM_FMTBIT_S32_LE);
+
+	if (ret < 0)
+		goto out;
+
+out:
+	return ret;
+}
+
+static struct snd_soc_ops broxton_gpmrb_tdf8532_ops = {
+	.startup = broxton_gpmrb_tdf8532_startup,
+};
 /* broxton digital audio interface glue - connects codec <--> CPU */
 static struct snd_soc_dai_link broxton_gpmrb_dais[] = {
 	/* Front End DAI links */
@@ -223,6 +253,7 @@ static int broxton_gpmrb_hdmi_startup(struct snd_pcm_substream *substream)
 		.trigger = {SND_SOC_DPCM_TRIGGER_POST,
 			    SND_SOC_DPCM_TRIGGER_POST},
 		.dpcm_playback = 1,
+		.ops = &broxton_gpmrb_tdf8532_ops,
 	},
 	[BXT_AUDIO_TUNER_CP] = {
 		.name = "Dirana Tuner Cp Port",
-- 
1.9.1

